x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -127,33 +138,61 @@ private static void prepareEcsDefinitions() throws IOException {
+            bodyBuilder.startObject();
+            bodyBuilder.startObject("settings");
+            bodyBuilder.field("index.mapping.total_fields.limit", 10000);
+            bodyBuilder.endObject();
+            bodyBuilder.field("mappings", ecsDynamicTemplates);
+            bodyBuilder.endObject();
+
+            Request createIndexRequest = new Request("PUT", "/" + indexName);
+            createIndexRequest.setJsonEntity(Strings.toString(bodyBuilder));
+            // noinspection resource
+            Response response = ESRestTestCase.client().performRequest(createIndexRequest);
+            Assert.assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+        }
+        // ensureGreen(indexName);
 
         Map<String, Object> flattenedFieldsMap = new HashMap<>();
         for (Map.Entry<String, Map<String, Object>> fieldEntry : ecsFlatFieldDefinitions.entrySet()) {
             String fieldName = fieldEntry.getKey();
             Map<String, Object> fieldDefinitions = fieldEntry.getValue();
             String type = (String) fieldDefinitions.get("type");
-            assertNotNull(
+            Assert.assertNotNull(
                 String.format(Locale.ENGLISH, "Can't find type for field '%s' in %s file", fieldName, ECS_DYNAMIC_TEMPLATES_FILE),
                 type
             );
             Object testValue = generateTestValue(type);
             flattenedFieldsMap.put(fieldName, testValue);
         }
-        IndexResponse indexResponse = index("test-flattened-fields", "1", flattenedFieldsMap);
-        assertEquals(indexResponse.getResult(), DocWriteResponse.Result.CREATED);
 
-        GetMappingsResponse mappingsResponse = indicesAdmin().prepareGetMappings("test-flattened-fields").get();
-        assertThat(mappingsResponse.mappings().size(), equalTo(1));
-        Map<String, Object> mappings = mappingsResponse.mappings().entrySet().iterator().next().getValue().getSourceAsMap();
-        assertTrue(mappings.containsKey("properties"));
+        try (XContentBuilder bodyBuilder = JsonXContent.contentBuilder()) {
+            Request indexRequest = new Request("POST", "/" + indexName + "/_doc");
+            indexRequest.setJsonEntity(Strings.toString(bodyBuilder.map(flattenedFieldsMap)));
+            // noinspection resource
+            Response response = ESRestTestCase.client().performRequest(indexRequest);
+            Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusLine().getStatusCode());
+        }
+
+        Request getMappingRequest = new Request("GET", "/" + indexName + "/_mapping");
+        // noinspection resource
+        Response response = ESRestTestCase.client().performRequest(getMappingRequest);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
+        Map<String, Object> mappingResponse;
+        try (
+            XContentParser parser = XContentFactory.xContent(XContentType.JSON)
```

Comment:
A parser for json can be constructed as follows:

```
XContentParser parser = createParser(JsonXContent.jsonXContent, response.getEntity().getContent());
```

Reply:
I cannot use it from a static context though...

Reply:
Why do you need a static context? This seems to be in a non-static test?